.github/workflows/test_on_push.yml

```diff
@@ -122,7 +121,8 @@ jobs:
         if: (matrix.os == 'ubuntu-latest' && matrix.python-version != 3.11) || (matrix.os != 'ubuntu-latest')
         with:
           path: ${{ github.workspace }}/.nox/unit/
-          key: ${{ runner.os }}-nox-unit-${{ steps.setup-python.outputs.python-version }}-${{ hashFiles('**/noxfile.py', '**/setup.py') }}
+          key: ${{ runner.os }}-nox-unit-${{ steps.setup-python.outputs.python-version }}-${{ hashFiles('**/setup.py') }}
+          restore-keys: ${{ runner.os }}-nox-unit-${{ steps.setup-python.outputs.python-version }}-
```

Comment:
How are the restore-keys different from keys? Or how is this improving/helping the cache system?

Reply:
In cases wherein no cache hits occur on a specific key hash (which has been the case a lot lately), it falls back to using of the listed `restore-keys` sequentially to restore files from. Sort of a _partial_ match. Here we only use one restore key because the Python version should always be in it

Reply:
Also, would the `pip` cache work if we were to install "-e .[all]" in the `setup-python` step? (I think it would if we install standard Python dependencies and PyBaMM-specific onew in that step itself, see the end of [this section](https://github.com/actions/setup-python/blob/v4/docs/advanced-usage.md#caching-packages). That way all `nox` environments can benefit from it as well

Reply:
I think it should work.